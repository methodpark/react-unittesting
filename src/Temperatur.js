import React from 'react';

export default class Temperatur extends React.Component {
  render() {
    const {label, value} = this.props;

    return (
      <div id={label.toLowerCase()} className="temp">
        {label}: {this._round(value)}Â°C
      </div>
    );
  }

  _round(value) {
    return Math.round(value * 100) / 100;
  }
}