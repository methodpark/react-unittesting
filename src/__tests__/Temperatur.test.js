import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Temperatur from '../Temperatur';

// ------------------- rendering without jest --------
const component = <Temperatur 
  label='label'
  value={42} />;

// shallow rendering
const shallowWrapper = shallow(component);

// full dom rendering
const fullDOMWrapper = mount(component);

// static rendered
const staticRendered = render(component);



const resultByTag = shallowWrapper.find('div');
console.log(resultByTag.text()); // "label: 42°C"

const resultById = shallowWrapper.find('#label');
console.log(resultById.text()); // "label: 42°C"

const resultByClass = shallowWrapper.find('.temp');
console.log(resultByClass.text());

const resultByMisc = shallowWrapper.find('div.temp');
console.log(resultByMisc.text());



describe('render', () => {
  it('should render the correct id', () => {
    const label = "Temperatur";
    const temperature = 42.229;

    const wrapper = shallow(<Temperatur 
      label={label} value={temperature} />);

    const id = wrapper.find('div').prop('id');
    const expected = 'temperatur';
    expect(id).toEqual(expected);
  });

  it('should render the temperature and label', () => {
    const label = "Temperatur";
    const temperature = 42.229;

    const wrapper = shallow(<Temperatur 
      label={label} value={temperature} />);
      
    const text = wrapper.find('div').text();
    const expected = 'Temperatur: 42.23°C';
    expect(text).toEqual(expected);
  });
});

describe('snapshot', () => {
  it('should match the snapshot', () => {
    const label = "Temperatur";
    const temperature = 42;

    const rendered = shallow(<Temperatur 
      label={label} value={temperature} />);

    expect(rendered).toMatchSnapshot();
  });
});