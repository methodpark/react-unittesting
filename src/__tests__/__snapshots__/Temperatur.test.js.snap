// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should match the snapshot 1`] = `
<div
  className="temp"
  id="temperatur"
>
  Temperatur
  : 
  42
  Â°C
</div>
`;
